test/e2e/support/utils/set-up-response-mocking.js

```diff
@@ -21,15 +31,23 @@
  * @param {Array} mocks Array of mock settings.
  */
 export async function setUpResponseMocking( mocks ) {
-	await page.setRequestInterception( true );
-	page.on( 'request', async ( request ) => {
-		for ( let i = 0; i < mocks.length; i++ ) {
-			const mock = mocks[ i ];
-			if ( mock.match( request ) ) {
-				await mock.onRequestMatch( request );
-				return;
+	if ( ! interceptionInitialized ) {
+		// We only want to set up the request interception once, or else we get a crash
+		// when we try to process the same request twice.
+		interceptionInitialized = true;
+		await page.setRequestInterception( true );
+		page.on( 'request', async ( request ) => {
```

Comment:
https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#class-page

I see `page.once` in this section. Could that be used instead?

Reply:
`once` is for responding to an event once, not only registering a handler once, unfortunately, and we want our handler to be there for all requests.

Reply:
> we want our handler to be there for all requests.

All requests... within a particular file of tests? Or the entire suite?

To me, this request mocking should only take place for the duration of the tests where it's applicable. It seems we're missing a critical "tear down" step, which risks lingering test state inadvertently affecting other test files.